Code/GraphMol/ForceFieldHelpers/UFF/Builder.cpp

```diff
@@ -84,48 +91,39 @@ boost::uint8_t getTwoBitCell(boost::shared_array<boost::uint8_t> &res,
 // ------------------------------------------------------------------------
 boost::shared_array<boost::uint8_t> buildNeighborMatrix(const ROMol &mol) {
   unsigned int nAtoms = mol.getNumAtoms();
-  unsigned nTwoBitCells = (nAtoms * nAtoms - 1) / 4 + 1;
+  unsigned nTwoBitCells = (nAtoms * (nAtoms + 1) - 1) / 8 + 1;
   boost::shared_array<boost::uint8_t> res(new boost::uint8_t[nTwoBitCells]);
-  for (unsigned int i = 0; i < nTwoBitCells; ++i) {
-    res[i] = 0;
-  }
   for (unsigned int i = 0; i < nAtoms; ++i) {
-    unsigned int iTab = i * nAtoms;
     for (unsigned int j = i; j < nAtoms; ++j) {
-      setTwoBitCell(res, iTab + j, RELATION_1_X);
-      setTwoBitCell(res, i + j * nAtoms, RELATION_1_X);
+      setTwoBitCell(res, twoBitCellPos(nAtoms, i, j), RELATION_1_X);
```

Comment:
This double loop is a lot of extra indexing work just to do what is equivalent to a memset for the entire array. Should be able to replace this double loop with the following:

```
memset(&res[0], RELATION_1_X, nTwoBitCells);
```